assets/src/blocks/SocialMedia/SocialMediaEditorScript.js

```diff
@@ -0,0 +1,244 @@
+import { useEffect } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+  RadioControl,
+  SelectControl,
+  PanelBody,
+} from '@wordpress/components';
+import { SocialMediaEmbed } from './SocialMediaEmbed';
+import { URLInput } from '../../components/URLInput/URLInput';
+import { HTMLSidebarHelp } from '../../components/HTMLSidebarHelp/HTMLSidebarHelp';
+import { fetchJson } from '../../functions/fetchJson';
+import {
+  OEMBED_EMBED_TYPE,
+  FACEBOOK_EMBED_TYPE,
+  FACEBOOK_PAGE_TAB_TIMELINE,
+  FACEBOOK_PAGE_TAB_EVENTS,
+  FACEBOOK_PAGE_TAB_MESSAGES,
+  ALLOWED_OEMBED_PROVIDERS,
+} from './SocialMediaConstants.js';
+
+const FB_API_BASE_URL  = 'https://graph.facebook.com/v9.0';
+const INSTAGRAM_OEMBED = `${FB_API_BASE_URL}/instagram_oembed`;
+const FB_ACCESS_TOKEN = window.p4ge_vars.planet4_options.fb_app_access_token;
```

Comment:
Is this token intended to be distributed publicly? It's entered in the settings here:
https://github.com/greenpeace/planet4-master-theme/blob/a872eb5a98dd58e45da522be48704805ebc845ac/src/Settings.php#L281-L284

Reply:
That line was added for this ticket: https://jira.greenpeace.org/browse/PLANET-5840

Reply:
Their is a link to the documentation about the value that is entered there, that page contains following text:

> Note that because this request uses your app secret, it must never be made in client-side code or in an app binary that could be decompiled. It is important that your app secret is never shared with anyone. Therefore, this API call should only be made using server-side code.

Reply:
Hmmm yes I see the problem 🤔 Should we use an API call here instead then?

Reply:
I'm afraid so, I don't think the tokens can have only read access, since their documentation states (a bit further down)

> Again, for security, app access token should never be hard-coded into client-side code, doing so would give everyone who loaded your webpage or decompiled your app full access to your app secret, and therefore the ability to modify your app.

Reply:
Facebook has a few ways to embed things, maybe we can look at alternatives? The way we're currently using also has some other limitations (for example we're loading the iframe twice because it's not responsive).

Could be we can also use "Client Access Tokens", which are intended to be shipped with clients. https://developers.facebook.com/docs/plugins/oembed#client-access-tokens

Reply:
This part of the code is used for the Instagram embed, not the Facebook one, even though the name might be confusing 😅

Reply:
Updated the code to use an endpoint for the embed code!

Reply:
Nice! Maybe @sagarsdeshmukh can tell whether we are already using the client token in the Planet 4 settings?

Reply:
Thanks @Pieter for point it 👍  Yes the "App Access Tokens" which is we are currently using is secure token and only recomended to use in backend (serverside code), as Pieter already mention If we want to consume a graph api in client side/frontend we need a "Client Access Tokens" which works exactly same.
@mleray - DM me and I will share you the "Client Access Tokens" :)